lib/publiccloud/provider.pm

```diff
@@ -441,21 +446,26 @@ Destroys the current terraform deployment
 =cut
 sub terraform_destroy {
     my ($self) = @_;
-    my $cmd;
     return if get_required_var('TEST') =~ /^publiccloud_upload_img/;
     record_info('INFO', 'Removing terraform plan...');
+
+    my $cmd = 'terraform destroy -no-color -auto-approve';
     if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
-        $cmd = 'terraform destroy -no-color -auto-approve';
+
+        # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
+        # On HA/SAP tests with ha-sap-terrafrom-deployments tool we shouldn't force the region for all providers
+        $cmd .= " -var 'aws_region=$self->region'" if is_ec2;
```

Comment:
I read you already did some verification runs with this version, so this may be minor, but I think it would be better to have `$self->region` outside of the double quotes as in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/provider.pm#L197.

Leaving it within the double quotes could lead to funny business:

```
susetest# perl
use Mojo::File qw(path);

my $a = path('/etc/os-release');
my $b = "FILE CONTENTS: [$a->slurp]\n";
print $b;
^D
FILE CONTENTS: [/etc/os-release->slurp]
susetest# perl
use Mojo::File qw(path);

my $a = path('/etc/os-release');
my $b = "FILE CONTENTS: [" . $a->slurp . "]\n";
print $b;
^D
FILE CONTENTS: [NAME="openSUSE Leap"
VERSION="15.1"
ID="opensuse-leap"
ID_LIKE="suse opensuse"
VERSION_ID="15.1"
PRETTY_NAME="openSUSE Leap 15.1"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:leap:15.1"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
]
```

Reply:
There is this interpolation syntax
```perl
my $b = "FILE CONTENTS: [${\ $a->slurp() }]";
```
which also works, but I'm not a fan of it :)

Reply:
@alvarocarvajald you're right, it's not good... But as @cfconrad said in fact this change is not needed for our HA/SAP tests, so I will simply revert these specific changes.

Reply:
>There is this interpolation syntax
>which also works, but I'm not a fan of it :)

Yeah, me neither :P

> @alvarocarvajald you're right, it's not good... But as @cfconrad said in fact this change is not needed for our HA/SAP tests, so I will simply revert these specific changes.

Oh, cool. Better yet.